pkg/apis/pytorch/v1/pytorchjob_types.go

```diff
@@ -43,26 +43,10 @@ type PyTorchJob struct {
 
 // PyTorchJobSpec is a desired state description of the PyTorchJob.
 type PyTorchJobSpec struct {
-	// Specifies the duration (in seconds) since startTime during which the job can remain active
-	// before it is terminated. Must be a positive integer.
-	// This setting applies only to pods where restartPolicy is OnFailure or Always.
-	// +optional
-	ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
-
-	// Number of retries before marking this job as failed.
-	// +optional
-	BackoffLimit *int32 `json:"backoffLimit,omitempty"`
-
-	// Defines the policy for cleaning up pods after the PyTorchJob completes.
-	// Defaults to None.
-	CleanPodPolicy *common.CleanPodPolicy `json:"cleanPodPolicy,omitempty"`
-
-	// Defines the TTL for cleaning up finished PyTorchJobs (temporary
-	// before Kubernetes adds the cleanup controller).
-	// It may take extra ReconcilePeriod seconds for the cleanup, since
-	// reconcile gets called periodically.
-	// Defaults to infinite.
-	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
+	// RunPolicy encapsulates various runtime policies of the distributed training
+	// job, for example how to clean up resources and how long the job can stay
+	// active.
+	RunPolicy *common.RunPolicy `json:"runPolicy,omitempty"`
```

Comment:
Should this change be included in v1? It may break existing users.

Reply:
The magic here is user already use this way using v1 crd. (because we don't have crd validation)

```
spec:
  runPolicy:
    cleanPodPolicy: All
```

If we generate crd from types.go now, it will block user to use above way so e2e tests failed.  

This change actually guarantee they can use above way. From users perspective, they don't have user facing changes

Reply:
Gotcha, LGTM